robot/client/client_test.go

```diff
@@ -2044,3 +2044,27 @@ func TestCloudMetadata(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, md, test.ShouldResemble, injectCloudMD)
 }
+
+func TestShutDown(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	listener, err := net.Listen("tcp", "localhost:0")
+	test.That(t, err, test.ShouldBeNil)
+	injectRobot := &inject.Robot{
+		ResourceNamesFunc:   func() []resource.Name { return nil },
+		ResourceRPCAPIsFunc: func() []resource.RPCAPI { return nil },
+		ShutdownFunc:        func(ctx context.Context) error { return nil },
+	}
+	gServer := grpc.NewServer()
+	pb.RegisterRobotServiceServer(gServer, server.New(injectRobot))
+	go gServer.Serve(listener)
+	defer gServer.Stop()
+
+	client, err := New(context.Background(), listener.Addr().String(), logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer func() {
+		test.That(t, client.Close(context.Background()), test.ShouldBeNil)
+	}()
+
+	err = client.Shutdown(context.Background())
+	test.That(t, err, test.ShouldBeNil)
```

Comment:
this doesn't properly test that the client sends a message to the server - we should do something like https://github.com/viamrobotics/rdk/blob/main/components/arm/server_test.go#L84-L87 and maybe toggle a bool to indicate we reached the Shutdown func